File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/stateengine/StackList.java
package org.firstinspires.ftc.teamcode.stateengine;

import org.firstinspires.ftc.teamcode.hardware.RobotHardware;

public abstract class StackList {

  private State[] stack;

  public void createStack(State[] stack) { this.stack = stack; }

  public State[] getStack() { return stack; }

  public void initStack(RobotHardware rh) {
    for (State state : getStack()) {
      state.init(rh);
    }
  }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/stateengine/StackList.kt
package org.firstinspires.ftc.teamcode.stateengine

import org.firstinspires.ftc.teamcode.hardware.RobotHardware

abstract class StackList(private val stack: Array<State>) {
  fun initStack(rh: RobotHardware) {
    for (state in this.stack) {
      state.init(rh)
    }
  }
}
